src/IB/FEMechanicsExplicitIntegrator.cpp

```diff
@@ -0,0 +1,739 @@
+
+        // Create FE equation systems objects and corresponding variables.
+        d_equation_systems[part].reset(new EquationSystems(*d_meshes[part]));
+        EquationSystems& equation_systems = *d_equation_systems[part];
+        d_fe_data[part]->setEquationSystems(d_equation_systems[part].get(),
+                                            /*level_number*/ 0); // TODO: remove level_number as a required parameter
+        if (from_restart)
+        {
+            const std::string& file_name = libmesh_restart_file_name(
+                d_libmesh_restart_read_dir, d_libmesh_restart_restore_number, part, d_libmesh_restart_file_extension);
+            const XdrMODE xdr_mode = (d_libmesh_restart_file_extension == "xdr" ? DECODE : READ);
+            const int read_mode =
+                EquationSystems::READ_HEADER | EquationSystems::READ_DATA | EquationSystems::READ_ADDITIONAL_DATA;
+            equation_systems.read(file_name,
+                                  xdr_mode,
+                                  read_mode,
+                                  /*partition_agnostic*/ true);
+        }
+        else
+        {
+            auto& X_system = equation_systems.add_system<ExplicitSystem>(COORDS_SYSTEM_NAME);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                X_system.add_variable("X_" + std::to_string(d), d_fe_order_position[part], d_fe_family_position[part]);
+            }
+            X_system.get_dof_map()._dof_coupling = &d_diagonal_system_coupling;
+
+            auto& dX_system = equation_systems.add_system<ExplicitSystem>(COORD_MAPPING_SYSTEM_NAME);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                dX_system.add_variable(
+                    "dX_" + std::to_string(d), d_fe_order_position[part], d_fe_family_position[part]);
+            }
+            dX_system.get_dof_map()._dof_coupling = &d_diagonal_system_coupling;
+
+            auto& U_system = equation_systems.add_system<ExplicitSystem>(VELOCITY_SYSTEM_NAME);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                U_system.add_variable("U_" + std::to_string(d), d_fe_order_position[part], d_fe_family_position[part]);
+            }
+            U_system.get_dof_map()._dof_coupling = &d_diagonal_system_coupling;
+
+            auto& F_system = equation_systems.add_system<ExplicitSystem>(FORCE_SYSTEM_NAME);
+            for (unsigned int d = 0; d < NDIM; ++d)
+            {
+                F_system.add_variable("F_" + std::to_string(d), d_fe_order_force[part], d_fe_family_force[part]);
+            }
+            F_system.get_dof_map()._dof_coupling = &d_diagonal_system_coupling;
+        }
+
+        setup_system_vectors(
+            &equation_systems, { COORDS_SYSTEM_NAME, VELOCITY_SYSTEM_NAME }, { "current", "half", "new" });
+        setup_system_vectors(
+            &equation_systems, { FORCE_SYSTEM_NAME }, { "current", "half", "new", "tmp", "RHS Vector" });
```

Comment:
This is the same code as in IBFEMethod. Could move this also to the base class and then both classes can use that (and IBFEMethod can then set up FEDataManager too)?

Reply:
I am not sure that they will ultimately use the same collections of vectors.

Reply:
It wouldn't be too hard to have a class set up whatever extra vectors it needs but at the moment they look like the same sets to me (`X`, `dX`, `U`, `F`).